services/powchain/src/main/java/tech/pegasys/teku/services/powchain/PowchainService.java

```diff
@@ -66,29 +67,50 @@ public class PowchainService extends Service {
   private final List<Web3j> web3js;
   private final OkHttpClient okHttpClient;
 
-  public PowchainService(final ServiceConfig serviceConfig, final PowchainConfiguration powConfig) {
-    checkArgument(powConfig.isEnabled());
+  public PowchainService(
+      final ServiceConfig serviceConfig,
+      final PowchainConfiguration powConfig,
+      final Optional<Web3j> trustedWeb3j,
+      final Optional<String> trustedEndpoint) {
+    checkArgument(powConfig.isEnabled() || trustedWeb3j.isPresent());
 
     AsyncRunner asyncRunner = serviceConfig.createAsyncRunner("powchain");
 
     this.okHttpClient = createOkHttpClient();
-    this.web3js = createWeb3js(powConfig);
     final SpecConfig config = powConfig.getSpec().getGenesisSpecConfig();
-    Eth1ProviderSelector eth1ProviderSelector =
-        new Eth1ProviderSelector(
-            IntStream.range(0, web3js.size())
-                .mapToObj(
-                    idx ->
-                        new Web3jEth1Provider(
-                            powConfig.getSpec().getGenesisSpecConfig(),
-                            serviceConfig.getMetricsSystem(),
-                            Eth1Provider.generateEth1ProviderId(
-                                idx + 1, powConfig.getEth1Endpoints().get(idx)),
-                            web3js.get(idx),
-                            asyncRunner,
-                            serviceConfig.getTimeProvider()))
-                .collect(Collectors.toList()));
-
+    final Eth1ProviderSelector eth1ProviderSelector;
+    if (trustedWeb3j.isPresent()) {
+      LOG.warn("Eth1 endpoint not provided, using execution engine endpoint for eth1 data");
```

Comment:
I don't think this is a warning - it's actually the scenario we expect post-merge. Might be worth logging at INFO level though.

Reply:
True, the same way I see it. It's already a mess for users to have two different endpoints for eth1